qcengine/programs/molpro.py

```diff
@@ -68,82 +70,115 @@ def compute(self, input_data: 'ResultInput', config: 'JobConfig') -> 'Result':
                     template: Optional[str] = None) -> Dict[str, Any]:
-        input_file = []
-        posthf_methods = {'mp2', 'ccsd', 'ccsd(t)'}
-
-        # Write header info
-        input_file.append("!Title")
-        # Memory is in megawords per core for Molpro
-        memory_mw_core = int(config.memory * (1024 ** 3) / 8e6 / config.ncores)
-        input_file.append("memory,{},M".format(memory_mw_core))
-        # TODO Add specification of wfu file only if asked for
-        # input_file.append('file,2,dispatch.wfu')
-        input_file.append('')
-
-        # Have no symmetry by default
-        input_file.append('{symmetry,nosym}')
-        input_file.append('')
-
-        # Write the geom
-        input_file.append('geometry={')
-        for sym, geom in zip(input_model.molecule.symbols, input_model.molecule.geometry):
-            s = "{:<4s} {:>{width}.{prec}f} {:>{width}.{prec}f} {:>{width}.{prec}f}".format(
-                sym, *geom, width=14, prec=10)
-            input_file.append(s)
-        input_file.append('}')
-
-        # Write charge and multiplicity
-        input_file.append('set,charge={}'.format(input_model.molecule.molecular_charge))
-        input_file.append('set,multiplicity={}'.format(input_model.molecule.molecular_multiplicity))
-        input_file.append('')
-
-        # Write the basis set
-        input_file.append('basis={')
-        input_file.append('default,{}'.format(input_model.model.basis))
-        input_file.append('}')
-        input_file.append('')
-
-        # Start of Molpro Commands
-        # Write energy call
-        write_hf = input_model.model.method.lower() in posthf_methods
-        if write_hf:
-            input_file.append('{HF}')
-        input_file.append('{{{:s}}}'.format(input_model.model.method))
-
-        # Write gradient call if asked for
-        if input_model.driver == 'gradient':
+        if template is None:
+            input_file = []
+            posthf_methods = {'mp2', 'ccsd', 'ccsd(t)'}
+
+            # Memory is in megawords per core for Molpro
+            memory_mw_core = int(config.memory * (1024 ** 3) / 8e6 / config.ncores)
+            input_file.append("memory,{},M".format(memory_mw_core))
+            # TODO Decide how I want this keyword to look/work
+            # if input_model.keywords.get('wfu') is not None:
+            #     input_file.append('file,2,{}'.format(input_model.keywords.get('wfu')))
+            input_file.append('')
+
+            # Don't orient the molecule if asked to fix_com or fix_orientation
+            if input_model.molecule.fix_orientation or input_model.molecule.fix_com:
+                input_file.append('{orient,noorient}')
+
+            # Have no symmetry if asked to fix_symmetry
+            if input_model.molecule.fix_symmetry == 'c1':
+                input_file.append('{symmetry,nosym}')
+            else:
+                input_file.append('{symmetry,auto}')
```

Comment:
As I see it, there are 3 things one could reasonably expect to get back from `to_string(dtype=...)`:
* (a) Cartesian coordinate lines only, with perhaps the Natom line for XYZ
* (b) All the molecule info that can be got out of the schema and the dtype program will accept _that can be handed to the dtype program **contiguous** to the Cartesian lines_
* (c) All the molecule info ... will accept with the non-contiguous or options-like parts returned in a dictionary.

I started off with (a). Currently most dtypes are (b). I'm doing (c) for cfour. Glad to gather opinions. The symmetry and orient could be included with (b) or (c).

Reply:
It looks like I've mostly done option (b) for the `to_string` method for Molpro. I think it makes sense that the `to_string` function with specific `dtype`s for each program handles all of the molecule info that is relevant for that program. Unsure about option (b) vs (c) though.

Reply:
Should I go ahead and make another PR for QCEl or should I wait for further discussion on this?

Reply:
I think, yes, go ahead and do another qcel PR for orient and symmetry. will need to be careful that the meanings map directly. isn't there a `symmetry,cs`, etc. in molpro for fixing C_s symmetry for an otherwise C_2v mol?

Reply:
That could very well be the case, unfortunately I'm not too familiar with the symmetry functionality in Molpro.

Reply:
This is now handled in the `to_string` method of `molecule`. PR https://github.com/MolSSI/QCElemental/pull/72

Reply:
Great, should have a new release tomorrow and will rerun the CI.

Once this goes through is there anything preventing a merge?

Reply:
I think everything should be good to go. Oh Lori had a question above about the memory assignment.